server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1440,7 +1440,7 @@ public void processPendingDeletes(Index index, IndexSettings indexSettings, Time
                         sleepTime = Math.min(maxSleepTimeMs, sleepTime * 2); // increase the sleep time gradually
                         logger.debug("{} schedule pending delete retry after {} ms", index, sleepTime);
                     }
-                } while ((System.nanoTime() - startTimeNS) < timeout.nanos());
+                } while ((System.nanoTime() - startTimeNS) < timeout.nanos() && lifecycle.closed() == false);
```

Comment:
I think we want to abort on stop, no need to wait until fully closed?

```suggestion
                } while ((System.nanoTime() - startTimeNS) < timeout.nanos() && lifecycle.started());
```